packages/payload/src/queues/operations/runJobs/index.ts

```diff
@@ -105,40 +106,38 @@ export const runJobs = async ({
   // the same job being picked up by another worker
   const jobsQuery: {
     docs: BaseJob[]
-  } = id
-    ? {
-        docs: [
-          (await req.payload.update({
+  } = {
+    docs: id
+      ? [
+          await updateJob({
             id,
-            collection: 'payload-jobs',
             data: {
               processing: true,
-              seenByWorker: true,
             },
             depth: req.payload.config.jobs.depth,
             disableTransaction: true,
-            showHiddenFields: true,
-          })) as BaseJob,
-        ],
-      }
-    : ((await req.payload.update({
-        collection: 'payload-jobs',
-        data: {
-          processing: true,
-          seenByWorker: true,
-        },
-        depth: req.payload.config.jobs.depth,
-        disableTransaction: true,
-        limit,
-        showHiddenFields: true,
-        where,
-      })) as unknown as PaginatedDocs<BaseJob>)
+            req,
+            returning: true,
+          }),
+        ]
+      : ((await updateJobs({
+          data: {
+            processing: true,
+          },
+          depth: req.payload.config.jobs.depth,
+          disableTransaction: true,
+          limit,
+          req,
+          returning: true,
+          where,
+        })) ?? []),
+  }
```

Comment:
Again your code is hard to read. 20+ line ternaries with the same values assigned in some cases is sloppy. This is not as easy to maintain as it could be.